[JS] Storage API - path update (#7145)
aws-amplify/docs: src/pages/[platform]/build-a-backend/storage/list/index.mdx

<InlineFilter filters={['javascript', 'react-native', 'angular', 'nextjs', 'react', 'vue']}>

> You can list all or paginated set of objects uploaded under a given path.

Reviewer: ```suggestion
You can either list all objects uploaded under a given path or opt to receive a paginated list of objects.
```

      add(elements.join('/'), target[element], item);
    }
  };
  response.items.forEach((item) => add(item.path, filesystem, item));
  return filesystem;
}
```

This places each item's data inside a special `__data` key.

## Access all files

To get a list of all files in your S3 bucket under a specific prefix, you can set `listAll` to `true`.

<BlockSwitcher>
<Block name="With Path (Above v6.x.x)">
```javascript
import { list } from 'aws-amplify/storage';

try {
  const response = await list({

>     path: 'photos/'

Reviewer: ```suggestion
    path: 'photos/',
```

  console.log('Error ', error);
}
```
</Block>
</BlockSwitcher>

## Paginated file access

If the pageSize is set lower than the total file size, a single `list` call only returns a subset of all the files. To list all the files with multiple calls, users can use the `nextToken` flag:

<BlockSwitcher>
<Block name="With Path (Above v6.x.x)">
```javascript
import { list } from 'aws-amplify/storage';

const PAGE_SIZE = 20;
let nextToken = undefined;
//...
const loadNextPage = async () => {
  let response = await list({

>     path: 'photos/'

Reviewer: ```suggestion
    path: 'photos/',
```

    // Alternatively, path: ({identityId}) => `album/{identityId}/photos/`

>       pageSize: PAGE_SIZE,

Reviewer: Tabbing inconsistency
